packages/eslint-plugin/src/rules/no-inputs-metadata-property.ts

```diff
@@ -29,6 +30,26 @@ export default createESLintRule<Options, MessageIds>({
       } ${Selectors.metadataProperty(METADATA_PROPERTY_NAME)}`](
         node: TSESTree.Property,
       ) {
+        const ancestorMayBeHostDirectiveAPI = node.parent?.parent?.parent;
```

Comment:
question: I see a difference between the implementation here https://github.com/angular-eslint/angular-eslint/pull/1231/files#diff-34e0047dec14410dccd7fee53b69b0edeb6c62c8b6d4480a4d6144e95aa323b5. What's the reason for it?

Reply:
Only a slight difference 😄 . So in `no-input-rename`, the targeted selector is `Selectors.INPUTS_METADATA_PROPERTY_LITERAL`, while the targeted selector for this rule is `${
        Selectors.COMPONENT_OR_DIRECTIVE_CLASS_DECORATOR
      } ${Selectors.metadataProperty(METADATA_PROPERTY_NAME)}`. Although either of the two selectors *could* work for this rule, the main reason why I kept the latter selector is because when we do report a lint violation, we want to highlight the `inputs` keyword like shown in the invalid test cases. When using the former selector, we'd end up highlighting the value bound to `inputs`, which I think is less helpful.